sdk/data/azappconfig/client_test.go

```diff
@@ -1242,3 +1242,243 @@ func TestTagsFilteringEndToEnd(t *testing.T) {
+				if strings.HasPrefix(*setting.Key, "setting-") {
+					foundKeys = append(foundKeys, *setting.Key)
+				}
+			}
+		}
+
+		// Should find only the setting with service=api
+		require.Contains(t, foundKeys, "setting-with-normal-tags")
+		require.NotContains(t, foundKeys, "setting-with-nil-tag-value")   // service is nil
+		require.NotContains(t, foundKeys, "setting-with-empty-tag-value") // service is empty
+	})
+
+	// Test 3: Filter by empty string tag value
+	t.Run("EmptyStringTagValueFilter", func(t *testing.T) {
+		selector := azappconfig.SettingSelector{
+			TagsFilter: []string{"service="},
+		}
+
+		var foundKeys []string
+		pager := client.NewListSettingsPager(selector, nil)
+		for pager.More() {
+			page, err := pager.NextPage(context.Background())
+			require.NoError(t, err)
+
+			for _, setting := range page.Settings {
+				if strings.HasPrefix(*setting.Key, "setting-") {
+					foundKeys = append(foundKeys, *setting.Key)
+				}
+			}
+		}
+
+		// Should find settings where service tag has empty string value
+		require.Contains(t, foundKeys, "setting-with-empty-tag-value")
+		require.NotContains(t, foundKeys, "setting-with-normal-tags")   // service=api
+		require.NotContains(t, foundKeys, "setting-with-nil-tag-value") // service is nil
+	})
+
+	// Test 4: Complex filter with multiple conditions
+	t.Run("ComplexMultipleTagsFilter", func(t *testing.T) {
+		selector := azappconfig.SettingSelector{
+			TagsFilter: []string{"env=staging", "service=web", "version=2.0"},
+		}
+
+		var foundKeys []string
+		pager := client.NewListSettingsPager(selector, nil)
+		for pager.More() {
+			page, err := pager.NextPage(context.Background())
+			require.NoError(t, err)
+
+			for _, setting := range page.Settings {
+				if strings.HasPrefix(*setting.Key, "setting-") {
+					foundKeys = append(foundKeys, *setting.Key)
+				}
+			}
+		}
+
+		// Should find only setting-mixed-tags that matches all conditions
+		require.Contains(t, foundKeys, "setting-mixed-tags")
+		require.Len(t, foundKeys, 1) // Only one setting should match all conditions
+	})
+
+	// Test 6: Filter with non-existent tag
+	t.Run("NonExistentTagFilter", func(t *testing.T) {
+		selector := azappconfig.SettingSelector{
+			TagsFilter: []string{"nonexistent=value"},
+		}
+
+		var foundKeys []string
+		pager := client.NewListSettingsPager(selector, nil)
+		for pager.More() {
+			page, err := pager.NextPage(context.Background())
+			require.NoError(t, err)
+
+			for _, setting := range page.Settings {
+				if strings.HasPrefix(*setting.Key, "setting-") {
+					foundKeys = append(foundKeys, *setting.Key)
+				}
+			}
+		}
+
+		// Should find no settings with non-existent tag
+		require.Empty(t, foundKeys)
+	})
+
+	// Test 7: Filter by tag key that exists with nil values
+	t.Run("TagKeyExistsWithNilValuesFilter", func(t *testing.T) {
+		selector := azappconfig.SettingSelector{
+			TagsFilter: []string{"tag1=\x00"}, // tag1 exists with nil or empty values in some settings
```

Comment:
What happens if you pass "tag1=\0"? I know it's interchangeable since both are URL encoded to "%00", but I'm curious if we prefer one over the other in Go SDK? We normally publicize "\0" as the filter for null value in our docs and sample code.